### ISSUE TITLE
Problem with a logical expression

### ISSUE THREAD

**Issue by ontana** [neo4j/neo4j#12051]
Hi guys,
I am facing a problem with the neo4j cypher in version 3.4.8. I am running the following query:

```
MATCH (company:Company) 
WHERE company.status IN ["ACTIVE"] AND NOT company.status IN ["DELETED"] 
RETURN company
```

On

- Neo4j version: 3.3.1-enterprise
- Operating system: Ubuntu 16.04
- Connecting through [http://localhost:7474/browser/]
- It returns a list of companies correctly.

However,
On
- Neo4j version: 3.4.8-enterprise
- Operating system: Ubuntu 16.04
- Connecting through [http://localhost:7474/browser/]
- It returns empty.

When change that cypher:
```
MATCH (company:Company) 
WHERE company.status IN ["ACTIVE"] AND company.status <> "DELETED" 
RETURN company
```

It worked fine.

I know the cypher is not best practise, but it's just an example for my case. Is this a bug? Or am I just missing something about the Cypher's syntax?

**Comment by sherfert** [comment]
Hi @ontana thanks for the report.
This seems to be a bug that was introduced in the 3.4 series.

**Comment by ontana** [comment]
Hi @sherfert, thank you for your reply. Looking forward to hearing you soon.

**Comment by Lojjs** [comment]
@ontana Thank you so much for your bug report. We have now merged a fix which will be included in the next 3.4 patch release (3.4.10) as well as our upcoming 3.5 release.

Best regards Louise, Neo4j Cypher team
